src/types/api/main.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export type DesktopAPI = {
+
+    // Initialization
+    isDev: () => Promise<boolean>;
+    getAppInfo: () => Promise<{name: string; version: string}>;
+    reactAppInitialized: () => void;
+
+    // Session
+    setSessionExpired: (isExpired: boolean) => void;
+    onUserActivityUpdate: (listener: (
+        userIsActive: boolean,
+        idleTime: number,
+        isSystemEvent: boolean,
+    ) => void) => void;
+
+    // Unreads/mentions/notifications
+    sendNotification: (title: string, body: string, channelId: string, teamId: string, url: string, silent: boolean, soundName: string) => void;
```

Comment:
I actually wanted to be much more prescriptive about what we expect to be sent up.
We actually did have these in an object of sorts while doing message passing but that was just an `any` as I mentioned.

I can see the benefit of compatibility where we just add/remove fields we support/don't support, but again I wanted to be more prescriptive. If we want to help the caller, I would prefer we just add in some TSDocs so that they know what does which, I think that would be more helpful.

Reply:
I think it's fine to be prescriptive but still make it an object since you can just make the fields required. I care mostly about there being 7 arguments, most of which are strings which are easy to confuse the order of since it's easier to screw up
```js
sendNotification(
    'title',
    'body',
    'myteam', // team and channel are swapped because this function has them backwards in the web app
    'mychannel',
    'url',
);
```
than it is to screw up
```js
sendNotification({
    title: 'title',
    body: 'body',
    teamId: 'myteam',
    channelId: 'mychannel',
    url: 'url',
});
```
because order no longer matters. Even if we document it, no one reads that anyway 😛

Reply:
I'd say it's just as easy to fat-finger it one way or the other, plus the latter has more code. I'm 3/5 on keeping it the way it is. I'd also rather not invite anyone to try and shove anything else into that object without proper typing, or hacking stuff in any way.

Reply:
I don't agree that it's as easy to make a mistake with an object parameter because the parameter names are clearly visible instead of the reader having to remember the exact order of them, but that's fine if you want to leave them as-is